README.md

```diff
@@ -7,32 +7,136 @@ Inspired from [Caddy](https://github.com/mholt/caddy).
 ## Features
 
 * Easy configuration with Maddyfile (same as Caddyfile)
-* Automatic TLS and DKIM support
 * Transparent PGP support
 * Runs anywhere with no external dependencies (not even libc)
 
-## Usage
+## Installation
 
 ```shell
-go get github.com/emersion/maddy/...
+go get github.com/emersion/maddy/cmd/maddy
 ```
 
-Create a file named `Maddyfile`:
+Build tags:
+* `nosqlite3`
+  Disable SQLite3 support in go-sqlmail (enabled by default). Saves around 9
+  MiB of binary size.
+* `mysql`
+  Include support for MySQL driver in go-sqlmail.
+* `postgresql`
+  Include support for PostgreSQL driver in go-sqlmail.
 
+## Configuration
+
+#### Syntax
+
+Maddy uses configuration format similar (but not the same!) to Caddy's
+Caddyfile.  You may want to read [Caddyfile page from Caddy docs](https://caddyserver.com/docs/caddyfile).
+
+Notable differences from Caddy's format:
+* Directives can't span multiple lines
+* Environmental variables are not expanded
+* You can't omit braces if you have only one configuration block
+
+#### Modularity
+
+Maddy does have a module-based design and you specify in configuration modules
+you want to use and how you want them to work.
+
+Generic syntax for configuration block is as follows:
 ```
-imaps://127.0.0.1:1993 {
-	proxy imaps://mail.example.org
-	tls self_signed
-	compress
-	pgp
+module_name instance_name {
+    configuration_directives
 }
+```
+
+`instance_name` is the unique name of the configuration block. It is used when
+you need to refer to the module from different place in configuration (e.g.
+configure SMTP to deliver mail to certain specific storage)
+
+You can omit `instance_name`. If there is only one module config. block - it
+will get name the same as `module_name`. If there are multiple config blocks
```

Comment:
Hmm, maybe just disallow having multiple modules without a name? It becomes very difficult to track down which module instance is which, especially with includes.

Reply:
Right, will change to allow only one instance with "implicit" name.